### ISSUE TITLE
keyboard & switch

### ISSUE THREAD

**Issue by aplsimple** [rdbende/Forest-ttk-theme#2]
Hi, rdbende!

Perhaps, most annoying issue of the themes: in fact they are unusable with keyboard because there is no highlighting the focused widgets. And you never know which button is focused if there are several of them, so "don't press Enter/Space" is a rule.

Second issue is that the Switch widget seems to be unusable in forest theme, as "ttk::style layout Switch.TCheckbutton" isn't present in it at all.

I've tried to modify your theme to overcome these issues. The result is seen in the attached videos. The "before" video refers to the original theme, "after" to the modified one. Note that the keyboard is only used, the mouse being inactive.

For your consideration, the source (**themes.zip**) is attached too. The forest theme has undergone the most modifications, as seen in its video. I'd even changed main colors, but it's only my taste, never mind.

Take this stuff for a humble base to make your themes better.

Thanks a lot for your work, it's so well made!

With respect,
Alex

https://user-images.githubusercontent.com/37333988/131718128-09416526-9215-4e9d-9108-663ffb515985.mp4

https://user-images.githubusercontent.com/37333988/131718159-e3672ed8-da15-4562-a5e8-56a32ee101da.mp4

https://user-images.githubusercontent.com/37333988/131718189-ead9535f-259a-4a0b-982a-8a8a7425efce.mp4

[themes.zip](https://github.com/rdbende/Forest-ttk-theme/files/7093278/themes.zip)

**Comment by rdbende** [comment]
Hi Alex!

Unlike in my other themes, there is no `Switch.TCheckbutton` but a plain `Switch` instead. (I renamed it in my other themes for backwards compatibility with other themes, also `ToggleButton`, and `Card`). Probably this theme will have them too, because I have many other local fixes to it that I haven't pushed so far.

And keyboard focus... I didn't really deal with that, because as soon as I started my first theme I realized how ugly the built-in focus is and so I didn't implement it. In the Sun Valley theme, I thought about it, but whatever image I set for it, I could only achieve the dotted line, in different colors. While the original Microsoft design has much nicer focus highlights.

I'm wondering that you might be able to give advice on this, because I see you know Tcl/Tk really well.

**Comment by rdbende** [comment]
Fyi [Sun Valley ttk theme](https://github.com/rdbende/Sun-Valley-ttk-theme) finally got keyboard focus indicators, and I have to say I really like it.
I hope someday the Forest theme will get them too, and it will certainly be sooner than the Azure theme (tbh, I don't really like the Azure theme).

**Comment by aplsimple** [comment]
Good to know.

BTW I read your reasons to move to gitlab. Not quite sure of your decision.

**Comment by rdbende** [comment]
> BTW I read your reasons to move to gitlab. Not quite sure of your decision.

Yeah. While I still have concerns about Github, in the end I kept using it for pretty much everything, because I had to realize that I can't just simply move to Gitlab. I have quite a few popular projects, and I am a contributor to several repos on Github, so I can't just simply leave it.
